kcb.sh doesn't pass on the quotes to the Java CLI, fails with the new scope flag to parse --scope="openid profile offline_token"

### Describe the bug

kcb.sh doesn't pass on the quotes to the Java CLI, fails with the new scope flag to parse --scope="openid profile offline_token"


### Version

nightly (9/16/2022)

### Expected behavior

kcb.sh should parse the multi subject word of scope successfully and pass it to the Java CLI.

### Actual behavior

kcb.sh doesn't pass on the quotes to the Java CLI, fails with the new scope flag to parse --scope="openid profile offline_token"

```
$ ./kcb.sh   --scenario=keycloak.scenario.authentication.AuthorizationCode --server-url={KEYCLOAK_SERVER_URL}   --realm-name=test-realm --scope="openid profile offline_token"
Error: Could not find or load main class profile
Caused by: java.lang.ClassNotFoundException: profile

```

with the debug flag, running against a minikube setup:

```
+ case "`uname`" in
++ uname
++ readlink -f ./kcb.sh
+ RESOLVED_NAME=/home/kakella/code/keycloak-benchmark/benchmark/target/keycloak-benchmark-0.8-SNAPSHOT/bin/kcb.sh
+ '[' x/home/kakella/code/keycloak-benchmark/benchmark/target/keycloak-benchmark-0.8-SNAPSHOT/bin/kcb.sh = x ']'
+ GREP=grep
++ dirname /home/kakella/code/keycloak-benchmark/benchmark/target/keycloak-benchmark-0.8-SNAPSHOT/bin/kcb.sh
+ DIRNAME=/home/kakella/code/keycloak-benchmark/benchmark/target/keycloak-benchmark-0.8-SNAPSHOT/bin
+ JAVA_OPTS=-server
+ JAVA_OPTS='-server -Xmx1G -XX:+HeapDumpOnOutOfMemoryError'
+ DEBUG_MODE=false
+ DEBUG_PORT=8787
+ CONFIG_ARGS=
+ SCENARIO=keycloak.scenario.authentication.ClientSecret
+ '[' 4 -gt 0 ']'
+ case "$1" in
+ SCENARIO=keycloak.scenario.authentication.AuthorizationCode
+ shift
+ '[' 3 -gt 0 ']'
+ case "$1" in
+ [[ --server-url=http://keycloak.192.168.39.218.nip.io = --* ]]
+ CONFIG_ARGS=' -Dserver-url=http://keycloak.192.168.39.218.nip.io'
+ shift
+ '[' 2 -gt 0 ']'
+ case "$1" in
+ [[ --realm-name=test-realm = --* ]]
+ CONFIG_ARGS=' -Dserver-url=http://keycloak.192.168.39.218.nip.io -Drealm-name=test-realm'
+ shift
+ '[' 1 -gt 0 ']'
+ case "$1" in
+ [[ --scope=openid profile offline_token = --* ]]
+ CONFIG_ARGS=' -Dserver-url=http://keycloak.192.168.39.218.nip.io -Drealm-name=test-realm -Dscope=openid profile offline_token'
+ shift
+ '[' 0 -gt 0 ']'
+ '[' false = true ']'
+ CLASSPATH_OPTS='/home/kakella/code/keycloak-benchmark/benchmark/target/keycloak-benchmark-0.8-SNAPSHOT/bin/../lib/*'
+ exec java -server -Xmx1G -XX:+HeapDumpOnOutOfMemoryError -Dserver-url=http://keycloak.192.168.39.218.nip.io -Drealm-name=test-realm -Dscope=openid profile offline_token -cp /home/kakella/code/keycloak-benchmark/benchmark/target/keycloak-benchmark-0.8-SNAPSHOT/bin/../lib/keycloak-benchmark-0.8-SNAPSHOT.jar io.gatling.app.Gatling -bf /home/kakella/code/keycloak-benchmark/benchmark/target/keycloak-benchmark-0.8-SNAPSHOT/bin -rf /home/kakella/code/keycloak-benchmark/benchmark/target/keycloak-benchmark-0.8-SNAPSHOT/bin/../results -s keycloak.scenario.authentication.AuthorizationCode
Error: Could not find or load main class profile
Caused by: java.lang.ClassNotFoundException: profile


```

### How to Reproduce?

run against an existing Keycloak server with the below command

```
./kcb.sh   --scenario=keycloak.scenario.authentication.AuthorizationCode --server-url={KEYCLOAK_SERVER_URL}   --realm-name=test-realm --scope="openid profile offline_token"
```

### Anything else?

_No response_

--- a/benchmark/src/main/content/bin/kcb.sh
+++ b/benchmark/src/main/content/bin/kcb.sh
@@ -1,18 +1,18 @@
 #!/bin/bash
 
-case "`uname`" in
+case "$(uname)" in
     CYGWIN*)
-        CFILE = `cygpath "$0"`
-        RESOLVED_NAME=`readlink -f "$CFILE"`
+        CFILE=$(cygpath "$0")
+        RESOLVED_NAME=$(readlink -f "$CFILE")
         ;;
     Darwin*)
-        RESOLVED_NAME=`readlink "$0"`
+        RESOLVED_NAME=$(readlink "$0")
         ;;
     FreeBSD)
-        RESOLVED_NAME=`readlink -f "$0"`
+        RESOLVED_NAME=$(readlink -f "$0")
         ;;
     Linux)
-        RESOLVED_NAME=`readlink -f "$0"`
+        RESOLVED_NAME=$(readlink -f "$0")
         ;;
 esac
 
@@ -21,7 +21,7 @@ if [ "x$RESOLVED_NAME" = "x" ]; then
 fi
 
 GREP="grep"
-DIRNAME=`dirname "$RESOLVED_NAME"`
+DIRNAME=$(dirname "$RESOLVED_NAME")
 
 JAVA_OPTS="-server"
 JAVA_OPTS="${JAVA_OPTS} -Xmx1G -XX:+HeapDumpOnOutOfMemoryError"
@@ -29,7 +29,8 @@ JAVA_OPTS="${JAVA_OPTS} -Xmx1G -XX:+HeapDumpOnOutOfMemoryError"
 DEBUG_MODE="${DEBUG:-false}"
 DEBUG_PORT="${DEBUG_PORT:-8787}"
 
-CONFIG_ARGS=${CONFIG_ARGS:-""}
+CONFIG_ARGS=()
+SERVER_OPTS=()
 
 SCENARIO="keycloak.scenario.authentication.ClientSecret"
 
@@ -38,7 +39,7 @@ do
     case "$1" in
       --debug)
           DEBUG_MODE=true
-          if [ -n "$2" ] && [ "$2" = `echo "$2" | sed 's/-//'` ]; then
+          if [ -n "$2" ] && [ "$2" = "${2//-}" ]; then
               DEBUG_PORT=$2
               shift
           fi
@@ -55,10 +56,10 @@ do
           break
           ;;
       *)
-          if [[ $1 = --* || ! $1 =~ ^-.* ]]; then
-            CONFIG_ARGS="$CONFIG_ARGS -D${1:2}"
+          if [[ $1 = --* || ! $1 =~ ^-D.* ]]; then
+            CONFIG_ARGS+=("-D${1:2}")
           else
-            SERVER_OPTS="$SERVER_OPTS $1"
+            SERVER_OPTS+=("$1")
           fi
           ;;
     esac
@@ -67,7 +68,7 @@ done
 
 # Set debug settings if not already set
 if [ "$DEBUG_MODE" = "true" ]; then
-    DEBUG_OPT=`echo $JAVA_OPTS | $GREP "\-agentlib:jdwp"`
+    DEBUG_OPT=$(echo "$JAVA_OPTS" | $GREP "\-agentlib:jdwp")
     if [ "x$DEBUG_OPT" = "x" ]; then
         JAVA_OPTS="$JAVA_OPTS -agentlib:jdwp=transport=dt_socket,address=$DEBUG_PORT,server=y,suspend=y"
     else
@@ -77,4 +78,4 @@ fi
 
 CLASSPATH_OPTS="$DIRNAME/../lib/*"
 
-exec java $JAVA_OPTS $CONFIG_ARGS -cp $CLASSPATH_OPTS io.gatling.app.Gatling -bf $DIRNAME -rf "$DIRNAME/../results" -s $SCENARIO
+exec java $JAVA_OPTS "${SERVER_OPTS[@]}" "${CONFIG_ARGS[@]}" -cp $CLASSPATH_OPTS io.gatling.app.Gatling -bf $DIRNAME -rf "$DIRNAME/../results" -s $SCENARIO

